### PR TITLE
Change the "select player" selectors to a text fields

### PR DESCRIPTION
**Pull request by kevincardonag**
image

### CODE REVIEW

File: components/game/api/types.ts [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
+  Move,
+  Player,
+  PlayerName,
+  PossibleCoords,
+} from "../types";
 
 export interface GameResponse {
   field: Field;
-  currentPlayer: Player;
+  currentPlayer: PlayerName;
```
Comment (dearlordylord) [inline components/game/api/types.ts]: just a note, a bit cleaner from a data perspective would be to leave it 1/2 (or an enum) here

File: components/game/api/useInitGame.ts [hunk 2]
```diff
@@ -1,13 +1,17 @@
 import { useCallback, useState } from "react";
-import { GameId } from "../types";
+import { GameId, PlayerName } from "../types";
 
 export const useInitGame = () => {
   const [loading, setLoading] = useState(false);
-  const init = useCallback(async () => {
+  const init = useCallback(async (PlayerOneName: PlayerName) => {
```
Comment (dearlordylord) [inline components/game/api/useInitGame.ts]: var name needs to be in the same style as other code, i.e. camelCase is very usual for js projects

File: components/game/lib.ts [hunk 3]
```diff
@@ -128,22 +137,30 @@ export class Game {
...
+      : (this.playerTwoName as PlayerName);
   }
 
   areCoordsValid(coords: Coords) {
     return !!this.possibleCoords().find(
       (cs) => cs.x === coords.x && cs.y === coords.y
     );
   }
+
+  setPlayerName(playerName: PlayerName, player: number) {
```
Comment (dearlordylord) [inline components/game/lib.ts]: minor note, arg order should be player -> playerName, same as with `Map.set(key, value)`

File: components/home.tsx [hunk 4]
```diff
@@ -1,65 +1,87 @@
...
-      router.push(`/game/${gameId}/${player}`);
+    (
+      gameId: GameId,
+      playerOneName?: PlayerName,
+      playerTwoName?: PlayerName
+    ) => {
+      if (playerOneName !== undefined) {
+        router.push(`/game/${gameId}/${playerOneName}`);
+      } else if (playerTwoName !== undefined) {
+        const res = fetch("/api/game/setPlayerTwo", {
```
Comment (dearlordylord) [inline components/home.tsx]: note1: would player2 be able to join again?

note2: would this mean that player2 name can be changed but player1 name cannot?

File: components/home.tsx [hunk 5]
```diff
@@ -1,65 +1,87 @@
...
+          method: "POST",
+          body: JSON.stringify({
+            playerTwo: playerTwoName,
+            gameId: gameId,
+          }),
+          headers: {
+            "Content-Type": "application/json",
+          },
+        });
+        if (!res.ok) {
```
Comment (dearlordylord) [inline components/home.tsx]: Is the expression res.ok always false at this point?

File: components/home.tsx [hunk 6]
```diff
@@ -1,65 +1,87 @@
...
-  const [playerForExistingGameInput, setPlayerForExistingGame] = useState(
-    `${PLAYER1}`
-  );
-  const [playerForNewGameInput, setPlayerForNewGame] = useState(`${PLAYER1}`);
+  const [playerForExistingGameInput, setPlayerForExistingGame] = useState("");
+  const [playerForNewGameInput, setPlayerForNewGame] = useState("");
   const handlePlayerForNewGameChange = useCallback(
-    (e: React.ChangeEvent<HTMLSelectElement>) => {
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      console.log(e.currentTarget.value);
```
Comment (dearlordylord) [inline components/home.tsx]: extra log

**Review by MiltonLn — APPROVED** [review]

File: components/home.tsx (hunk 4 above)
Comment (kevincardonag) [reply components/home.tsx]: yes, it is what you say, However the python implementation has the same behavior, so I would like to know @MiltonLn's opinion, should  I refactor it or take another approach?

Comment (MiltonLn) [reply components/home.tsx]: @kevincardonag just try to replicate the behavior from the Python side, we want them to be exactly the same

**Review by dearlordylord — APPROVED** [review]

**Pull request merged**